elastic-mapping-updater/pom.xml

```diff
@@ -0,0 +1,229 @@
+                            <goal>verify</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <runOrder>alphabetical</runOrder>
+                    <systemPropertyVariables>
+                        <docker.host.address>${docker.host.address}</docker.host.address>
+                    </systemPropertyVariables>
+                    <environmentVariables>
+                        <CAF_SCHEMA_UPDATER_ELASTIC_HOSTNAMES>${docker.host.address}</CAF_SCHEMA_UPDATER_ELASTIC_HOSTNAMES>
+                        <CAF_SCHEMA_UPDATER_ELASTIC_REST_PORT>${es.http.port}</CAF_SCHEMA_UPDATER_ELASTIC_REST_PORT>
+                        <CAF_SCHEMA_UPDATER_ELASTIC_CONNECT_TIMEOUT>5000</CAF_SCHEMA_UPDATER_ELASTIC_CONNECT_TIMEOUT>
+                        <CAF_SCHEMA_UPDATER_ELASTIC_SOCKET_TIMEOUT>60000</CAF_SCHEMA_UPDATER_ELASTIC_SOCKET_TIMEOUT>
+                        <CAF_LOG_LEVEL>INFO</CAF_LOG_LEVEL>
+                    </environmentVariables>
+                </configuration>
+            </plugin>
+
+            <plugin>
+                <groupId>io.fabric8</groupId>
+                <artifactId>docker-maven-plugin</artifactId>
+                <executions>
+                    <!-- Start the containers in pre-integration-test phase. -->
+                    <execution>
+                        <id>start-docker-containers</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>start</goal>
+                        </goals>
+                    </execution>
+                    <!-- Stop the containers in post-integration-test phase. -->
+                    <execution>
+                        <id>stop-docker-containers</id>
+                        <phase>post-integration-test</phase>
+                        <goals>
+                            <goal>stop</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <imagePullPolicy>IfNotPresent</imagePullPolicy>
+                    <useColor>false</useColor>
+                    <verbose>true</verbose>
+                    <removeVolumes>true</removeVolumes>
+                    <images>
+                        <!-- Elasticsearch image -->
+                        <image>
+                            <name>cafinternal/prereleases:opensuse-elasticsearch7-2.0.0-SNAPSHOT</name>
+                            <alias>elasticsearch</alias>
+                            <run>
+                                <env>
+                                    <discovery.type>single-node</discovery.type>
+                                </env>
+                                <ports>
+                                    <port>${es.http.port}:9200</port>
+                                    <port>${es.client.port}:9300</port>
```

Comment:
Why are there default values here, should these not be defined as properties at top level?  I notice the same values are added as a profile below too.

Reply:
if tests when run with the "use-default-fixed-ports" profile the the test containers can be accessed at the preset ports rather than the docker assigned ports.

Reply:
So if the profile or the property is not set, then there will be no value in the "es.http.port" property, and it will default to 9200, I understand this.  Just below this in the <wait>...</wait>  the "es.http.port" property is used again without a default value.   If "<es.http.port>9200</es.http.port>" is added as a property at top level this will not be an issue.  I hope this makes this clearer.

Reply:
From the dev build logs:
Here is the wait statement which just uses the docker assigned port "33155":

[INFO] DOCKER> [cafinternal/prereleases:opensuse-elasticsearch7-2.0.0-SNAPSHOT] "elasticsearch": Waited on url http://16.73.33.139:33155 12460 ms